OpenOversight/tests/routes/test_officer_and_department.py

```diff
@@ -186,18 +236,66 @@ def test_admin_can_edit_officer_badge_number(mockdata, client, session):
 
         assert 'Edited officer assignment' in rv.data.decode('utf-8')
         assert '<td>12345</td>' in rv.data.decode('utf-8')
-        assert officer.assignments[0].star_no == '12345'
+        assert '2019-02-01' in rv.data.decode('utf-8')
+        assert '2019-11-30' in rv.data.decode('utf-8')
 
+        assert officer.assignments[0].star_date == date(2019, 2, 1)
+        assert officer.assignments[0].resign_date == date(2019, 11, 30)
```

Comment:
This is similar to what I mentioned above, nice! A little shorter too, just need to make sure that there are no other assignments to this officer, but that should be fine I think

Reply:
Yea, in other tests I was just deleting other assignments as a first step, is that alright? Or do you want me to switch these to filtering for that assignment specifically?

Reply:
Eh I think actually calling the assignment and testing its attributes is better code. I'm not including the line `assert assignment is not None` if that's cool bc I think the following assertions imply that. Is that alright?

Reply:
you are right, it is implied. I thought maybe it helps finding the problem if the test later fails, but really the python error message is pretty clear anyways (like `AttributeError: 'NoneType' object has no attribute 'star_date'`). So yeah, good call skipping that one